metrics/README.md

```diff
@@ -0,0 +1,44 @@
+# Description
+
+This package configures Prometheus metrics for the node.
+
+# Technical Details
+
+We use a trick to combine our metrics with Geth ones.
+
+The `NewMetricsServer()` function in [`metrics.go`](./metrics.go) calls our own `Handler()` function which in turn calls two handlers:
+
+* `promhttp.HandlerFor()` - Our own custom metrics from this package.
+* `gethprom.Handler(reg)` - Geth metrics defined in [`metrics`](https://github.com/ethereum/go-ethereum/tree/master/metrics)
+
+By calling both we can extend existing metrics.
+
+# Metrics
+
+We add a few extra metrics on top of the normal Geth ones in [`node/metrics.go`](./node/metrics.go):
+
+* `p2p_peers_count` - Current numbers of peers split by name.
```

Comment:
Some of these metrics are redundant, for example `p2p_peers_count`, `p2p_peers_absolute`, and `p2p_peers_changes` carry the same information but provide different labels.

Reply:
Yeah, that's because I was copying legacy stuff. Do you think the `absolute` and `changes` metrics are useless? If so we can get rid of them and the `computeMetrics()` method.

Reply:
I think we can get rid of `_changes`. `_absolute` has this advantage over `_count` that it counts everything while `_count` parses names and skips peers. So only `_changes` seem to be really obsolete.

Reply:
Good point. I'll drop it.